### ISSUE TITLE
Flexibility in Java platform / minimum bytecode

### ISSUE THREAD

**Issue by cipherboy** [rpminspect/rpminspect#124]
From a recent package build for RHEL 7:

```json
...
   "java-bytecode": [
     {
       "result": "BAD",
       "waiver authorization": "Anyone",
       "message": "File \/org\/mozilla\/jss\/util\/NativeProxy.class (\/usr\/lib\/java\/jss4.jar), Java byte code version 52 greater than supported major version 51 for product release el7"
     },
     ...
   ],
...
```

The trick is that this package was built for the `rhel-x.y-java-unsafe-candidate` target tag, which gives Java 8 availability in the BUILDROOT. Packages can opt-in (especially, in a layered product) for building with these, in which case byte code version 52 should be the minimum, not version 51.

This tag doesn't change the dist tag (it is still `el7`), so some other mechanism for making an informed decision needs to be added.

**Comment by dcantrell** [comment]
If you look in lib/types.h at struct koji_build you will see what I get back from the Koji XML-RPC call.  One thing of value there is the task_id.  There's the struct koji_task which contains everything I get from getTaskInfo which I use if you ask rpminspect to compare modules (grrr).  That doesn't give much more information and I haven't yet found a way to get the Koji target via XML-RPC.  You can use the xmlrpc command line program and give it the Koji hub URL then the XML-RPC call and parameters as a list.  The Koji API is in doc/koji-api.txt (also "koji list-api" I think).  If you can figure out a way to get the build target via XML-RPC, that would help a lot.  I could further extend the Java major version mapping and allow target specification in addition to dist tag.

**Comment by dcantrell** [comment]
I think the best way we're going to have to handle this is by creating profiles in the rpminspect data package(s) for specific build targets like that.  Then we would modify the rpminspect runner so it knows to invoke builds originating from a specific Koji target with a specific rpminspect profile.  We're doing this now for things like the kernel and other large packages, but the modification is on the integration side in the rpminspect runner and not in the program itself.

I'm going to close this bug here and when the situation presents itself in one of the CI environments, I can work with the owner(s) of the rpminspect integration piece (the runner) and get things modified to account for situations like this.

TL;DR -- I believe enough configuration options are present now to construct the necessary configuration aspects in rpminspect to handle these situations.  The only requirement would be to modify the runner on the integration side.  That will be different for Fedora and RHEL, but that is the logical place since those Koji environments have entirely different build targets and other settings.
